doc/releases/index.rst

```diff
@@ -24,6 +24,7 @@ specific release and can be found at https://docs.zephyrproject.org/.
 .. toctree::
    :maxdepth: 1
 
+   release-notes-1.14
```

Comment:
One day we should consider adding the leading zero to these files e.g., release-notes-1.07, and we could use a ``:glob:`` and ``release-notes-*``) and not need to edit this file each release.  Alas, with the current file names, the release notes TOC won't sort correctly.